src/server/redis_connection.h

```diff
@@ -111,6 +111,7 @@ class Connection : public EvbufCallbackBase<Connection> {
   Worker *Owner() { return owner_; }
   void SetOwner(Worker *new_owner) { owner_ = new_owner; };
   int GetFD() { return bufferevent_getfd(bev_); }
+  bool HasRunningCommand() const { return saved_current_command_ != nullptr; }
```

Comment:
I'm wondering if there are non-blocking commands which are running in some situation?

Reply:
Yes, that's possible. 

For example,  connection0 is running on worker0 with the command `Get`, but it will wait for the lock if the migrating connection process is ongoing, then connection0 will be migrated to a new worker. Since then the new worker will start schedule the connection0 and the old worker will be still using the connection while resuming the ExecuteCommands.

Reply:
Oh, I mean, only blocking commands will be stored in `saved_current_command_ `.

Reply:
Yes, I also misunderstood what you mean. You're right, it should only have the data race in current_command, and the other variables should have the same issue.

Reply:
Fixed